ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -7487,6 +7487,81 @@ void HqlCppTranslator::doBuildAssignExecuteWhen(BuildCtx & ctx, const CHqlBoundT
     }
 }
 
+void HqlCppTranslator::doBuildExprPow(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & tgt)
+{
+    assertex(expr->numChildren() == 2);
+    
+    IHqlExpression * right = expr->queryChild(1);
+
+    bool notIntegerPower = !right->queryType()->isInteger();
+    if (notIntegerPower)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    IValue * value = right->queryValue();
+    bool notConstant = (value == nullptr);
+    if (notConstant) 
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    __int64 rightIntVal = value->getIntValue();
+    __int64 absExp = abs(rightIntVal);
+
+    const unsigned int MAX_EXP_INLINE_VALUE = 3;
+    if (absExp > MAX_EXP_INLINE_VALUE)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+    
+    OwnedITypeInfo realType = makeRealType(DEFAULT_REAL_SIZE);
+    CHqlBoundExpr leftBound;
+    
+    IHqlExpression * left = expr->queryChild(0);
+    OwnedHqlExpr castLeft = ensureExprType(left, realType);
+    buildTempExpr(ctx, castLeft, leftBound);
+
+    if (rightIntVal == 0)
+    {
+        tgt.expr.setown(createConstant(left->queryType()->castFrom(1.0)));
+    }
+    else
+    {
+        OwnedHqlExpr outExpr = LINK(leftBound.expr);
+        for (int i = 1; i < absExp; i++)
+        {
+            outExpr.setown(createValue(no_mul, LINK(realType), outExpr.getClear(), LINK(leftBound.expr)));
+        }
+
+        tgt.expr.setown(outExpr.getClear());
+
+        if (rightIntVal < 0)
+        {
+            OwnedHqlExpr numerator = createConstant(tgt.expr->queryType()->castFrom(1.0));
+            IHqlExpression* divisor = tgt.expr.get();
+            OwnedHqlExpr pureExpr = createValue(no_div, left->getType(), numerator.getClear(), LINK(divisor));
```

Comment:
bug: left->getType() is theoretically wrong - this should be realType (not sure if it currently can be non-real).